src/chunks/storage/array.rs

```diff
@@ -1,145 +1,147 @@
-#[cfg(feature = "memmap")]
-use std::fs::File;
-#[cfg(feature = "memmap")]
-use std::io::BufReader;
 use std::io::{Read, Seek, SeekFrom, Write};
 use std::mem::size_of;
 
-#[cfg(target_endian = "big")]
-use byteorder::ByteOrder;
 use byteorder::{LittleEndian, ReadBytesExt, WriteBytesExt};
-#[cfg(feature = "memmap")]
-use memmap::{Mmap, MmapOptions};
 use ndarray::{Array2, ArrayView2, ArrayViewMut2, CowArray, Ix1};
-#[cfg(feature = "memmap")]
-use ndarray::{Dimension, Ix2};
 
 use super::{Storage, StorageView, StorageViewMut};
-#[cfg(feature = "memmap")]
-use crate::chunks::io::MmapChunk;
 use crate::chunks::io::{ChunkIdentifier, ReadChunk, TypeId, WriteChunk};
 use crate::error::{Error, Result};
 use crate::util::padding;
 
-/// Memory-mapped matrix.
-#[derive(Debug)]
 #[cfg(feature = "memmap")]
-pub struct MmapArray {
-    map: Mmap,
-    shape: Ix2,
-}
+mod mmap {
+    use std::fs::File;
+    use std::io::{BufReader, Seek, SeekFrom, Write};
+    use std::mem::size_of;
+
+    use crate::chunks::io::MmapChunk;
+    #[cfg(target_endian = "big")]
+    use byteorder::ByteOrder;
+    use byteorder::{LittleEndian, ReadBytesExt};
+    use memmap::{Mmap, MmapOptions};
+    use ndarray::{ArrayView2, CowArray, Ix1};
+    use ndarray::{Dimension, Ix2};
+
+    use super::NdArray;
+    use crate::chunks::io::{ChunkIdentifier, TypeId, WriteChunk};
+    use crate::chunks::storage::{Storage, StorageView};
+    use crate::error::{Error, Result};
+    use crate::util::padding;
+
+    /// Memory-mapped matrix.
+    #[derive(Debug)]
+    #[cfg(feature = "memmap")]
```

Comment:
Isn't this redundant since it's contained in the already feature-gated `mmap` mod?

Reply:
Indeed, missed that one!